### DISCUSSION TITLE
Expo Modules, Opting for an Expo-less boilerplate, and life of an M1 dev...

### DISCUSSION THREAD

**Discussion by dragonmaster784** [infinitered/ignite#1865] [Q&A]
So I have walking the hard road with Ignite and M1, maybe I should say M1 and mobile development in general; Ignite and its' cli has actually been a blessing if anything 🔥

The one thing I have noticed though is that most of the Expo modules (yes, that especially means you EXLinearGradient) have been causing a major ruckus in the iOS/Xcode environment. Xcode won't complete a build before failing on one of the Expo modules. Is this expected behaviour and we must just stay away from Xcode all together, as the cli builds work, or is this perchance an issue of some sort that can be resolved with a PR?

I tested a vanilla RN project and was able to get it to successfully build in Xcode, so it got me wondering if there is a flag to completely exclude Expo when Igniting a project. Or does anyone have a strategy they use to ignite without Expo?

@jamonholmgren I am a huge fan of your work and only have major appreciation for everything you and your team have done so far, you guys are absolutely awesome! If your eyes ever find this thread, I saw on stream that you got one of those new shinies (M1 Max I think), and wondered how your experience has been so far. 
Would it ever be in the cards for you to put together a brief video or article on how your settled in to a stable developer environment with an M1? Or any solid path to achieve a usable environment for RN devs.

Thanks for reading everyone, and any other thoughts are welcomed!

**Comment by jamonholmgren** [comment]
@dragonmaster784 

Sorry about how long it took me to get to this! I'm actually answering on-stream (today's stream -- check YouTube, about 1:15 into the stream). Thanks so much for your kind words -- I really appreciate it!

So far, I've been able to use Ignited apps without trouble, even in Xcode with Expo. It sounds like there's an issue with your environment, for sure. It's been nearly a month since you asked this question, so perhaps you've already found and fixed it (update here if so!)

Here's a screenshot from an Xcode-built fresh Ignite app on my M1.

<img width="200" alt="image" src="https://user-images.githubusercontent.com/1479215/153658554-c91d7ea3-0699-481a-98e2-4820b7fe6070.png">


If not, maybe we could compare environments and figure out what's going on. My `npx ignite-cli doctor` is here:

```
System
  platform           darwin                             
  arch               arm64                              
  cpu                10 cores     Apple M1 Max          
  directory          rnlive       /Users/jh/Code/rnlive 

JavaScript (and globally-installed packages)
  node               16.13.1                                   /usr/local/bin/node    
  npm                8.1.0                                     /opt/homebrew/bin/npm  
    expo-cli         5.0.3                                                            
    npm              8.1.0                                                            
    pnpm             6.24.1                                                           
    typescript       4.5.5                                                            
  yarn               1.22.17                                   /opt/homebrew/bin/yarn 
  pnpm               6.24.1                                    /opt/homebrew/bin/pnpm 
                     <no pnpm global package info available>                          

Ignite
  ignite-cli         7.10.2       /Users/jh/.npm/_npx/e31027f3785124a8/node_modules/.bin/ignite      
  ignite src         build        /Users/jh/.npm/_npx/e31027f3785124a8/node_modules/ignite-cli/build 

Android
  java               1.8.0_312    /Users/jh/.sdkman/candidates/java/current/bin/java 
  android home       -            /Users/jh/Library/Android/sdk                      

iOS
  xcode              13.2.1     
  cocoapods          1.11.2       /opt/homebrew/bin/pod 
```
